packages/search-common/src/types.ts

```diff
@@ -29,6 +30,7 @@ export interface SearchResult {
 
 export interface SearchResultSet {
   results: SearchResult[];
+  totalCount: number;
```

Comment:
Is it helpful (to future search backend implementers) to make this an optional? So that we have a frontend interface that _can_ handle the absence of a total, and we don't _force_ everybody to always supply this no matter what it costs in terms of performance. Or is it just confusing

Reply:
Yeah, right now one could say that it isn't necessary. But introducing this option later is a breaking change. But I'm not really into writing my own paging component that works without pages. The one from material UI react doesn't support it.

Reply:
Tbh we need a proper solution for cursor paginated content that doesn't support jumping to pages. It's gonna be very tricky to reliably be able to provide total counts and offsets for some datasets in the future. Perhaps we can go for the breaking change later though, in order to not to block things too much.

Reply:
Yeah, I'm also open to write it cursor based, but it doesn't seems to match the UX requirements of having a paging control, instead of an endless list that loads while scrolling or a load more button. The latter would be a nice pattern too.

Reply:
I will just add that, in my experience (outside of maybe the main web search engines) I don't think I've ever encountered a paginated search experience where **a)** I actually had to get to the last page, but also **b)** where the last few pages actually worked as you would expect.  ...Precisely because result count is fuzzy.  And that's probably fine for now.